crypto/dsa/dsa_key.c

```diff
@@ -59,6 +59,52 @@ int ossl_dsa_generate_public_key(BN_CTX *ctx, const DSA *dsa,
     return ret;
 }
 
+/*
+ * Refer: FIPS 140-3 IG 10.3.A Additional Comment 1
+ * Perform a KAT by duplicating the public key generation.
+ *
+ * NOTE: This issue requires a background understanding, provided in a separate
+ * document; the current IG 10.3.A AC1 is insufficient regarding the PCT for
+ * the key agreement scenario.
+ *
+ * Currently IG 10.3.A requires PCT in the mode of use prior to use of the
+ * key pair, citing the PCT defined in the associated standard. For key
+ * agreement, the only PCT defined in SP 800-56A is that of Section 5.6.2.4:
+ * the comparison of the original public key to a newly calculated public key.
+ */
+static int dsa_keygen_knownanswer_test(DSA *dsa, BN_CTX *ctx,
+                                       OSSL_CALLBACK *cb, void *cbarg)
+{
+    int len, ret = 0;
+    OSSL_SELF_TEST *st = NULL;
+    unsigned char bytes[512] = {0};
+    BIGNUM *pub_key2 = BN_new();
+
+    if (!pub_key2)
+        return 0;
+
+    st = OSSL_SELF_TEST_new(cb, cbarg);
+    if (st == NULL)
+        goto err;
+
+    OSSL_SELF_TEST_onbegin(st, OSSL_SELF_TEST_TYPE_PCT_KAT,
+                               OSSL_SELF_TEST_DESC_PCT_DSA);
+
+    if (!ossl_dsa_generate_public_key(ctx, dsa, dsa->priv_key, pub_key2))
+        goto err;
+
+    len = BN_bn2bin(pub_key2, bytes);
```

Comment:
sizeof(bytes) is not used anywhere.... Same comment applies to next file..

Reply:
Hmmmm, not a bug just a problem waiting to happen...

Still easy to return an error if the buffer isn't large enough.

Reply:
You mean before it uses the buffer?

Reply:
Yep